### PR TITLE
Windows Compatibility

### PR DESCRIPTION
**Pull request by James0x57**
Adding windows compatibility - Please review =)

### CODE REVIEW

**Comment by justinbmeyer** [comment]
Please add testee back in for now so this doesn't get merged without it.

**Comment by James0x57** [comment]
$ npm install
and
$ grunt build
both went through without error after re-adding testee.

**Comment by matthewp** [comment]
Ok, I'll try running this through the Windows CI (appveyor)

**Comment by matthewp** [comment]
It's failing because we don't have browsers set up in Appveyor, i'm looking up how to do this.

File: main.js [hunk 1]
```diff
@@ -42,6 +42,8 @@
 		},
 		dir = function(uri){
 			var lastSlash = uri.lastIndexOf("/");
+			if(lastSlash === -1)
```
Comment (matthewp) [inline main.js]: Add a comment for why we are doing this.

File: main.js [hunk 2]
```diff
@@ -299,6 +303,24 @@ var makeSteal = function(System){
 				addProductionBundles.call(this);
 			}
 		};
+
+	// checks if we're running in node, then prepends the "file:" protocol if we are
+	var envPath = function(val) {
+		//next line is var we can use to check if it's running on windows. May use if specifying the "file:" protocol causes issues on *nix machines.
+		//var isInWindows = (typeof process !== "undefined" && typeof process.platform !== "undefined" && /^win/.test(process.platform));
```
Comment (matthewp) [inline main.js]: Get rid of this code, we know this works fine on unix.

**Comment by matthewp** [comment]
We should do a setter for `paths` as well.

**Comment by matthewp** [comment]
Yay, finally. There is a bug in npm v1.x that causes npm install to break on Windows some times, but happened often on AppVeyor. Upgrading to npm 2.x on AppVeyor fixes the issue and now we have a clean build.

**Comment by matthewp** [comment]
Oh, we need to get a path setter working with this as well. I'll update it.

**Pull request merged**